### PR TITLE
fixes 632 - BaselineProcessor test

### PR DESCRIPTION
**Pull request by rotty3000**
this adds adds simple workspace-less baseline processor, and workspace-less deployer

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java [hunk 1]
```diff
@@ -247,235 +253,6 @@ public Jar getLastRevision() throws Exception {
...
-	 * version is specified, we take the highest version with the same base
-	 * version.
-	 * <p>
-	 * Since baselining is expensive and easily generates errors you must enable
-	 * it. The easiest solution is to {@code -baseline: *}. This will match all
-	 * sub builders and will calculate the version.
-	 * 
-	 * @return a Jar or null
-	 */
-	public Jar getBaselineJar() throws Exception {
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java]: This is a public method. I would think you should leave this method here and change its implementation to call `baselineProcessor.getReleaseRepo()`.

Comment (rotty3000) [inline biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java]: I'm confused! the ProjectBuilder does indeed call down to the BaselineProcessor.

Comment (rotty3000) [inline biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java]: .. for that method.

Comment (rotty3000) [inline biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java]: See https://github.com/rotty3000/bnd/commit/28c4c9923e1a339e8de87aa03935875fa1a758fa

Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java]: OK. I see it now. It was moved so it did not appear next to the removal of the old method implementation. Those kind of changes make it harder on reviewers :-) I would be nice to not remove the javadoc comments too.

Comment (rotty3000) [inline biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java]: I can certainly update the PR with any changes you like!

So, I will:

* not separate the removal of the method from the move operation
* not remove the javadoc from the public method

OK?

**Comment by rotty3000** [comment]
Please see updates!

**Comment by bjhargrave** [comment]
> workspace-less deployer

Seems like the deployer still requires a workspace:

> public void deploy(String name, File file) throws Exception {
>  List<RepositoryPlugin> plugins = processor.getPlugins(RepositoryPlugin.class);

Where do you think the repository plugins come from? :-)

**Comment by rotty3000** [comment]
I'm not using a workspace!

I'm just using properties and this is working perfectly!

How the properties arrive is up to the implementer.

**Comment by bjhargrave** [comment]
But the concept of repositories comes from the workspace. I guess I am curious what a complete example looks like. Can you put a repo in github with a complete example using this?

**Comment by rotty3000** [comment]
If you want to see this code in use and working, please see the following
project where I've temporarily adapted the code to do this:

https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/BaselineJarTask.java#L153-L189

On Sun, Mar 8, 2015 at 2:56 PM, Raymond Auge <raymond.auge@liferay.com>
wrote:

> I'm not using a workspace!
>
> I'm just using properties and this is working perfectly!
>
> How the properties arrive is up to the implementer.
>



-- 
*Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
 (@rotty3000)
Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
 (@Liferay)
Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org> (@OSGiAlliance)

**Comment by rotty3000** [comment]
The plugin configurations arrives from here:

https://github.com/liferay/liferay-portal/blob/master/build.properties

Note this is not a bnd only project file.

which is passed into the BaselineProcessor here:

https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/BaselineJarTask.java#L157


On Sun, Mar 8, 2015 at 2:58 PM, Raymond Auge <raymond.auge@liferay.com>
wrote:

> If you want to see this code in use and working, please see the following
> project where I've temporarily adapted the code to do this:
>
>
> https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/BaselineJarTask.java#L153-L189
>
> On Sun, Mar 8, 2015 at 2:56 PM, Raymond Auge <raymond.auge@liferay.com>
> wrote:
>
>> I'm not using a workspace!
>>
>> I'm just using properties and this is working perfectly!
>>
>> How the properties arrive is up to the implementer.
>>
>
>
>
> --
> *Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
>  (@rotty3000)
> Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
>  (@Liferay)
> Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org>
> (@OSGiAlliance)
>



-- 
*Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
 (@rotty3000)
Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
 (@Liferay)
Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org> (@OSGiAlliance)

**Comment by rotty3000** [comment]
Repositories are not necessarily from workspaces.. that's just where they
have been traditionally configured. But if whomever is interested knows how
to set them up and handle the plugins, then who's to say they must have a
"workspace" to do that?

On Sun, Mar 8, 2015 at 3:01 PM, Raymond Auge <raymond.auge@liferay.com>
wrote:

> The plugin configurations arrives from here:
>
> https://github.com/liferay/liferay-portal/blob/master/build.properties
>
> Note this is not a bnd only project file.
>
> which is passed into the BaselineProcessor here:
>
>
> https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/BaselineJarTask.java#L157
>
>
> On Sun, Mar 8, 2015 at 2:58 PM, Raymond Auge <raymond.auge@liferay.com>
> wrote:
>
>> If you want to see this code in use and working, please see the following
>> project where I've temporarily adapted the code to do this:
>>
>>
>> https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/BaselineJarTask.java#L153-L189
>>
>> On Sun, Mar 8, 2015 at 2:56 PM, Raymond Auge <raymond.auge@liferay.com>
>> wrote:
>>
>>> I'm not using a workspace!
>>>
>>> I'm just using properties and this is working perfectly!
>>>
>>> How the properties arrive is up to the implementer.
>>>
>>
>>
>>
>> --
>> *Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
>>  (@rotty3000)
>> Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
>>  (@Liferay)
>> Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org>
>> (@OSGiAlliance)
>>
>
>
>
> --
> *Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
>  (@rotty3000)
> Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
>  (@Liferay)
> Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org>
> (@OSGiAlliance)
>



-- 
*Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
 (@rotty3000)
Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
 (@Liferay)
Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org> (@OSGiAlliance)

**Comment by bjhargrave** [comment]
> which is passed into the BaselineProcessor here: https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/BaselineJarTask.java#L157

The BaselineJarTask does not use a Deployer which is the new class you proposes which needs the plugins. I guessing you meant here: <https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/ReleaseToRepoTask.java#L62>.

> But if whomever is interested knows how to set them up and handle the plugins, then who's to say they must have a "workspace" to do that?

I guess the class doc needs to be more clear that for use outside the workspace, the argument to construct Deployers needs certain properties (-plugins, -pluginpath, -deployrepo). There was no test case for the standalone testing of Deployer to show this.

**Comment by rotty3000** [comment]
Baseline also needs the plugins because it "gets" bundles from repositories
to baseline against.

That ant task can optionally baseline against a jar file referenced
directly but that is not our common use case.

The deployer is here:

https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/ReleaseToRepoTask.java#L55-L69

It also needs the plugins obviously. It get's is configuration the same way
as the BaselineProcessor.

**Comment by rotty3000** [comment]
However, you're right that there isn't a deployment test.

For the record, there aren't any deployment tests to test the functionality
of Processors anywhere in the bnd project.

The only deployment test is directly on the implementation of the indexed
file repo.

On Sun, Mar 8, 2015 at 3:36 PM, Raymond Auge <raymond.auge@liferay.com>
wrote:

> Baseline also needs the plugins because it "gets" bundles from
> repositories to baseline against.
>
> That ant task can optionally baseline against a jar file referenced
> directly but that is not our common use case.
>
> The deployer is here:
>
>
> https://github.com/liferay/liferay-portal/blob/master/modules/sdk/ant-bnd/src/com/liferay/ant/bnd/ReleaseToRepoTask.java#L55-L69
>
> It also needs the plugins obviously. It get's is configuration the same
> way as the BaselineProcessor.
>



-- 
*Raymond Augé* <http://www.liferay.com/web/raymond.auge/profile>
 (@rotty3000)
Senior Software Architect *Liferay, Inc.* <http://www.liferay.com>
 (@Liferay)
Board Member & EEG Co-Chair, OSGi Alliance <http://osgi.org> (@OSGiAlliance)

**Pull request closed without merging**